### PR TITLE
Prevent potential underflows 

### LINKED ISSUE #5549: .size() -1 in for loops
We have a pattern where the unsigned loop variable in a for loop is compared against a vector ```.size() - 1```. This fails horribly when size is ```0```. This can be fixed by adding +1 on both sides of the comparison operator.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #5549

### CODE REVIEW

**Comment by jwpeterson** [comment]
:+1:  also, D:

File: framework/src/geomsearch/PenetrationThread.C [hunk 1]
```diff
@@ -216,9 +216,9 @@ PenetrationThread::operator() (const NodeIdRange & range)
 
         //Loop through all pairs of faces, and check for contact on ridge betweeen each face pair
         std::vector<RidgeData> ridgeDataVec;
-        for ( unsigned int i(0); i < p_info.size()-1; ++i )
+        for (unsigned int i = 0; i +1 < p_info.size(); ++i)
```
Comment (dschwen) [inline framework/src/geomsearch/PenetrationThread.C]: Oh, I forgot a space there.

**Comment by dschwen** [comment]
I'm changing a bit more code. There are a few ifs where this issue may come up. And gratuitous use of ```_some_variable[_some_variable.size()-1]``` which should be replaced by ```_some_variable.back()``` and a check that size() > 0

**Comment by dschwen** [comment]
Ugh, ```std::string.back()``` is C++11...

**Comment by jwpeterson** [comment]
> I could use *string.rbegin()

Sure, as long as `!string.empty()`.

**Comment by dschwen** [comment]
Yes, sure, back and rbegin do not solve the underflow issue, they are just
more idiomatic than the [size-1] construct.

**Pull request merged**